src/tbb/arena_slot.cpp

```diff
@@ -25,6 +25,15 @@ namespace r1 {
 //------------------------------------------------------------------------
 // Arena Slot
 //------------------------------------------------------------------------
+void arena_slot::reset_task_pool_and_leave(thread_data* td) {
+    td->my_arena->pool_mask[this_task_arena::current_thread_index()].store(0, std::memory_order_relaxed);
```

Comment:
```suggestion
    td->my_arena->pool_mask[td->my_arena_slot].store(0, std::memory_order_relaxed);
```
Can we use `td->my_arena_slot` instead of getting access to TLS via `this_task_arena::current_thread_index()`?